### DISCUSSION TITLE
Controller annotation cannot have char #

### DISCUSSION THREAD

**Discussion by ethanhere** [micronaut-projects/micronaut-core#10408] [Q&A]
A controller has a customized annotation. In the annotation class, the following line prevents the controller to be created as a bean:
 String foo() default "#{bar}";

Attached zip file is a maven project that reproduces the problem. I extended micronaut website created example by adding
SampleController.java: @Controller annoatated controller that is annotated by @SampleAnnotation
SampleAnnotation.java: my customized annotation class, line 18 (String foo() default "#{bar}";) is failing the test
SampleTest.java: just to test calling the controller.
 
If you unzip the zip file, you will have creating-your-first-micronaut-app-maven-java folder.
Commands to build:
cd creating-your-first-micronaut-app-maven-java
mvn clean install -DskipTests
Command to test:
mvn test -Dtest= SampleTest
 
The test would fail:
```
[ERROR]   SampleTest.sampletest:29 expected: <OK> but was: <NOT_FOUND>
The key log message that cause the failure:
09:20:00.416 [default-nioEventLoopGroup-1-3] DEBUG io.micronaut.http.server.netty.NettyRequestLifecycle - Request GET /samplecontroller
09:20:00.416 [default-nioEventLoopGroup-1-3] DEBUG io.micronaut.http.server.RequestLifecycle - No matching route: GET /samplecontroller

``` 
However, If you change line 18 of SampleAnnotation.java to
`String foo() default "!{bar}";`
The test would pass. The difference is that at 18 of SampleAnnotation.java char # is changed to char !
I also noticed the following log in the successful test:
```
09:26:54.700 [main] DEBUG io.micronaut.context.lifecycle - Created bean [example.micronaut.$SampleController$Definition$Intercepted@4351c8c3] from definition [Definition: example.micronaut.$SampleController$Definition$Intercepted] with qualifier [null]
…
09:26:54.747 [main] DEBUG io.micronaut.web.router.DefaultRouteBuilder - Created Route: GET /samplecontroller -> $SampleController$Definition$Intercepted#test (application/json)
09:26:54.747 [main] DEBUG io.micronaut.web.router.DefaultRouteBuilder - Created Route: HEAD /samplecontroller -> $SampleController$Definition$Intercepted#test (application/json)
 
```
Above 3 line of log cannot be found in the failing test log. In other words, the repro shows that ! char in annotation class is preventing the controller class created as a bean.
 
Can you example why this happens? The same code works in micronaut 3 (using char #). Can you tell me how to fix this?

**Comment by sdelamo** [comment]
Remove @Validated from your sample app controller and it will pass. # is used in evaluated expressions in Micronaut Framework 4.
